write.go

```diff
@@ -151,8 +151,18 @@ func (w *Writer) writeLiteral(l Literal) error {
 		}
 	}
 
-	_, err := io.Copy(w, l)
-	return err
+	// In case of bufio.Buffer, it will be 0 after io.Copy.
+	literalLen := int64(l.Len())
+
+	n, err := io.CopyN(w, l, literalLen)
+	if err != nil {
+		return err
+	}
+	if n != literalLen {
```

Comment:
`n == literalLen` is guaranteed by `io.CopyN`

Reply:
Not exactly, more like n <= literalLen.
```go
	foo := bytes.Buffer{}
	io.WriteString(&foo, "DIEEEEEEEEEEEEEE")

	fmt.Println(io.CopyN(ioutil.Discard, &foo, int64(len("DIEEEE"))))
	fmt.Println(foo.Len())
```
Will print `6 <nil>` and `10`.

Reply:
Is it fine for us that Literal may actually contain more data than Len() bytes?

Reply:
I don't think so, users will expect the whole `io.Reader` to be consumed. Users can always create an `io.LimitedReader` if needed.

Reply:
```
	foo := bytes.Buffer{}
	io.WriteString(&foo, "DIEEEEEEEEEEEEEE")
	literalLen := int64(len("DIEEEE")) // let's say foo.Len() returned this

	fmt.Println(io.CopyN(ioutil.Discard, &foo, literalLen))
	fmt.Println(foo.Len()) // some data is still left in buffer, will print 10
```

Reply:
Hmm? `io.CopyN` will return an error anyway right?

Reply:
It does not care about the actual amount of data in the reader, CopyN only guarantees that it will copy exactly N bytes passed in argument if there is more data than N - it will not return an error.

Reply:
Gah, yes, sorry I got confused!